### DISCUSSION TITLE
⭐ Configuring merge queue in your repo rulesets is now available in public beta!

### DISCUSSION THREAD

**Discussion by ettaboyle** [community/community#110908] [Announcements]
- Pull request in a merge queue will have an individual record in rule insights, making it easier to see who added the PR to the merge queue
- The rule insight dialog includes all pull requests that merged in the same group along with the checks needed for the queue

### Let us know what you think by reacting here or posting your feedback below. Thank you!

**Comment by Danajia2** [comment]
How can I perfectly operate in this channel?

**Reply by eleonard20224** [reply]
So far I like what I see in the latest update

**Comment by Lunacuartogibon** [comment]
Me gustaría aprender bien como funciona  y hacer las cosas al pie de la letra

**Reply by ettaboyle** [reply]
Hi @Lunacuartogibon, thanks for participating!

We take our [Code of Conduct](https://github.com/community/community/blob/0ca37cf594e438fd4aec3c7f3736b95e6b0b3567/CODE_OF_CONDUCT.md) very seriously and want to help ensure that everyone has a good experience free of antagonism and harassment. Unfortunately, we don’t currently have moderators for languages other than English. Until that changes, we need to ask that everyone use English here in the GitHub Community when posting. We’ll be locking any posts in languages other than English for now, including this one.

**Reply by pauleustice** [reply]
This seems like a strange response ^ Nothing in your CoC mentions posting in English, and you provide a translate function within the Discussions UI that makes it easy to understand non-English speakers. Coding is universal

**Comment by fitz-vivodyne** [comment]
When this goes live will there finally be a way to configure merge queues via API?

**Reply by patrick-knight** [reply]
Hey @fitz-vivodyne, we just release this to [general availability today](https://github.blog/changelog/2024-07-31-repository-updates-july-31st-2024/), and has support for merge queue via the rules REST and GraphQL APIs.

**Reply by fitz-vivodyne** [reply]
Awesome, thanks!

**Comment by usmonster** [comment]
Hello!

It seems that actors who can bypass other required branch protection rules (e.g. PR approval) can no longer bypass them if merge queue is required. In other words, there's no way to force adding a PR to the merge queue, **even when one has bypass rights for all repository rulesets**.

It might come from what's mentioned [in the doc](https://docs.github.com/en/enterprise-cloud@latest/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-a-merge-queue#about-merge-queues:~:text=Once%20a%20pull%20request%20has%20passed%20all%20required%20branch%20protection%20checks%2C):
> **Once a pull request has passed all required branch protection checks**, a user with write access to the repository can add the pull request to the queue.  

The current behavior feels like an overly strict interpretation of this, which tightly couples these required checks to the ability to use the merge queue without taking into account one's bypasser status.

_(FYI, it feels like there is a possible link to [this discussion around bypasser mergability status](https://github.com/orgs/community/discussions/61107#discussioncomment-7000915), though I haven't checked if this particular "required merge queue" issue also applies to the equivalent classic branch protection.)_

Our use case concerns a bot/app that we want to automatically generate and merge update PRs via the merge queue without human interaction. Our request is essentially the same as the one mentioned in [this comment](https://github.com/orgs/community/discussions/46757#discussioncomment-8455208): we want to allow some PRs to be added to the merge queue while bypassing other requirements. We could of course use our bypasser status to push the protected branch directly without a PR, but the goal is to open the PR and add it directly to the merge queue in order to avoid logical conflicts in our highly active repo.

**Is a solution to this issue on your radar?**

Thanks!

**Comment by novium2** [comment]
Retiring this post as it is over 6 months old. If you have any questions, consider starting a [new post](https://github.com/orgs/community/discussions/new/choose).
